src/integrationTest/java/uk/gov/hmcts/reform/cpo/repository/CasePaymentOrdersAuditTest.java

```diff
@@ -0,0 +1,151 @@
+package uk.gov.hmcts.reform.cpo.repository;
+
+import com.microsoft.applicationinsights.core.dependencies.apachecommons.lang3.RandomStringUtils;
+import org.hibernate.envers.AuditReader;
+import org.hibernate.envers.AuditReaderFactory;
+import org.hibernate.envers.RevisionType;
+import org.hibernate.envers.query.AuditEntity;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import uk.gov.hmcts.reform.cpo.BaseTest;
+import uk.gov.hmcts.reform.cpo.data.CasePaymentOrderEntity;
+import uk.gov.hmcts.reform.cpo.domain.CasePaymentOrder;
+import uk.gov.hmcts.reform.cpo.service.mapper.CasePaymentOrderMapper;
+
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.NoResultException;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+@SuppressWarnings({"PMD.JUnitAssertionsShouldIncludeMessage", "PMD.DataflowAnomalyAnalysis"})
+public class CasePaymentOrdersAuditTest extends BaseTest {
```

Comment:
Can't we use @DataJpaTest or something else instead of loading whole Spring Boot app. ?

Reply:
We could I guess, but I do like the simplicity of extending from the `BaseTest` for consistency, and for time being the app is small enough that I don't think it would have much impact on performance of tests?

Reply:
I thought it's DAO layer specific test. nw if setup is a bit complicated.